src/mesh.cpp

```diff
@@ -104,11 +104,33 @@ void scaleMesh(scene::IMesh *mesh, v3f scale)
 	for(u16 j=0; j<mc; j++)
 	{
 		scene::IMeshBuffer *buf = mesh->getMeshBuffer(j);
-		video::S3DVertex *vertices = (video::S3DVertex*)buf->getVertices();
-		u16 vc = buf->getVertexCount();
-		for(u16 i=0; i<vc; i++)
+		const video::E_VERTEX_TYPE vType = buf->getVertexType();
+		switch(vType)
 		{
-			vertices[i].Pos *= scale;
+			case video::EVT_STANDARD:
+				{
+				video::S3DVertex *vertices = (video::S3DVertex*)buf->getVertices();
+				u16 vc = buf->getVertexCount();
+				for(u16 i=0; i<vc; i++)
+					vertices[i].Pos *= scale;
+				}
+				break;
+			case video::EVT_2TCOORDS:
+				{
+				video::S3DVertex2TCoords *vertices = (video::S3DVertex2TCoords*)buf->getVertices();
+				u16 vc = buf->getVertexCount();
+				for(u16 i=0; i<vc; i++)
+					vertices[i].Pos *= scale;
+				}
+				break;
+			case video::EVT_TANGENTS:
+				{
+				video::S3DVertexTangents *vertices = (video::S3DVertexTangents*)buf->getVertices();
+				u16 vc = buf->getVertexCount();
```

Comment:
Consider echewing the abbreviation "vc" and using "vertex_count" instead or something more legible.